### PR TITLE
Implement the Clock service with a single method now

### PR DESCRIPTION
**Pull request by lindner**
- Also add MILLIS support for Instant date parsing/generation.
- Add toString to TimeUnit
- Tests

### CODE REVIEW

**Comment by lindner** [comment]
Addresses #3319

File: src/common/time/instant.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class Instant {
    * Returns the rounded number of milliseconds from the Epoch for this Instant..
    */
   get epochMilliseconds(): number {
-    return this.secs * 1000 + Math.round(this.nanos / 1000000);
+    return this.secs * 1000 + Math.round(this.nanos / 1000);
```
Comment (alxrsngrtn) [inline src/common/time/instant.ts]: Nice catch

File: src/common/time/instant.ts [hunk 2]
```diff
@@ -57,6 +57,27 @@ export class Instant {
     return this.resolutionValue;
   }
 
+  /**
+   * Returns a truncated Instant based on the current value and the
+   * specified TimeUnit.
+   *
+   * Throws if the specified TimeUnit is finer than the current TimeUnit;
```
Comment (alxrsngrtn) [inline src/common/time/instant.ts]: `@throws`

File: src/services/clock-service.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+      // Get current millis
+      const millis = Date.now();
+
+      // Truncate to timeunit and return
+      return Instant.fromEpochMilliseconds(millis).truncateTo(unit).toString();
+
+    } catch (e) {
+      // TODO document what happens during failures.
+      console.log('error ', e);
```
Comment (alxrsngrtn) [inline src/services/clock-service.ts]: Let's use the logger / import log.

File: src/services/tests/clock-test.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import {TimeUnit} from '../../common/time/timeunit.js';
+import '../clock-service.js';
+
+describe('services: clock', () => {
+  it('gets a clock value', async () => {
+    // SECONDS
+    let clockValue = await Services.request({call: 'clock.now', timeUnit: 'SECONDS'});
+    assert.isDefined(clockValue);
+
+    let instant = Instant.fromString(clockValue);
```
Comment (alxrsngrtn) [inline src/services/tests/clock-test.ts]: Will instant be importable / used from within particles as well?

File: src/common/time/instant.ts (hunk 2 above)
Comment (lindner) [reply src/common/time/instant.ts]: done

File: src/services/clock-service.ts (hunk 3 above)
Comment (lindner) [reply src/services/clock-service.ts]: done

File: src/services/tests/clock-test.ts (hunk 4 above)
Comment (lindner) [reply src/services/tests/clock-test.ts]: yes, it will be added to the default environment via loader.  PRs to follow.

**Pull request merged**